sdk/core/azure-core/src/main/java/com/azure/core/util/UserAgentUtil.java

```diff
@@ -0,0 +1,118 @@
+package com.azure.core.util;
+
+/**
+ * Utility for building user agent string for Azure client libraries as specified in the
+ * <a href="https://azure.github.io/azure-sdk/general_azurecore.html#telemetry-policy">design guidelines</a>.
+ */
+public final class UserAgentUtil {
+
+    public static final String DEFAULT_USER_AGENT_HEADER = "azsdk-java";
+
+    /*
+     * The base User-Agent header format is azsdk-java-<client_lib>/<sdk_version>. Additional information such as the
+     * application ID will be prepended and platform telemetry will be appended, a fully configured User-Agent header
+     * format is <application_id> azsdk-java-<client_lib>/<sdk_version> <platform_info>.
+     */
+    private static final String DEFAULT_USER_AGENT_FORMAT = DEFAULT_USER_AGENT_HEADER + "-%s/%s";
+
+    // From the design guidelines, the platform info format is:
+    // <language runtime>; <os name> <os version>
+    private static final String PLATFORM_INFO_FORMAT = "%s; %s; %s";
+
+    private UserAgentUtil() {
+        // don't instantiate
+    }
+
+    /**
+     * Return user agent string for the given sdk name and version.
+     *
+     * @param sdkName name of the SDK.
+     * @param sdkVersion Version of the SDK.
+     * @return User agent string as specified in design guidelines.
+     */
+    public static String toUserAgentString(String sdkName, String sdkVersion) {
```

Comment:
Thoughts on removing the `Configuration`-less APIs? If we need to extend these in the future to also add a property such as `String features` there would be the need for many additional APIs. Also, right now these additional APIs are only an internal convenience which we could handle with internal code.